### PR TITLE
[ticket/16823] Make datetime class correctly handle edge cases

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16823">PHPBB3-16823</a>.

### CODE REVIEW

File: phpBB/phpbb/datetime.php [hunk 1]
```diff
@@ -101,15 +101,15 @@ public function format($format = '', $force_absolute = false)
 				{
```
Comment (CHItA) [inline phpBB/phpbb/datetime.php:101]: shouldn't this be `if ($timestamp < $midnight + 2 * 86400)`

Comment (rxu) [reply phpBB/phpbb/datetime.php:101]: Good catch.

**Comment by rxu** [comment]
I'm not sure what to do with the 2nd part of the issue regarding handling `+5` / `+59` seconds ahead: either drop this behavior or leave it as is.

**Review by CHItA — APPROVED** [review]
